tests/postgres/test_command_ReadFile.py

```diff
@@ -0,0 +1,124 @@
+import pathlib
+import pytest
+import sqlalchemy
+from typing import Tuple
+
+from mara_app.monkey_patch import patch
+from mara_db import dbs, formats
+from mara_pipelines.commands.sql import ExecuteSQL
+from mara_pipelines.commands.files import ReadFile, Compression
+from tests.command_helper import run_command
+
+import mara_pipelines.config
+patch(mara_pipelines.config.data_dir)(lambda: pathlib.Path(__file__).parent)
+
+FILE_PATH = pathlib.Path(__file__).parent
+
+
+def db_is_responsive(db: dbs.DB) -> bool:
+    """Returns True when the DB is available on the given port, otherwise False"""
+    engine = sqlalchemy.create_engine(db.sqlalchemy_url, pool_pre_ping=True)
+
+    try:
+        with engine.connect() as conn:
+            return True
+    except:
+        return False
+
+
+@pytest.fixture(scope="session")
+def postgres_db(docker_ip, docker_services) -> Tuple[str, int]:
+    """Ensures that PostgreSQL server is running on docker."""
+
+    docker_port = docker_services.port_for("postgres", 5432)
+    mara_db = dbs.PostgreSQLDB(host=docker_ip,
+                               port=docker_port,
+                               user="mara",
+                               password="mara",
+                               database="mara")
+
+    # here we need to wait until the PostgreSQL port is available.
+    docker_services.wait_until_responsive(
+        timeout=30.0, pause=0.1, check=lambda: db_is_responsive(mara_db)
+    )
+
+    # create the dwh database
+    conn = dbs.connect(mara_db)  # dbt.cursor_context cannot be used here because
```

Comment:
Why no "with" here?

Reply:
`with` creates a transaction and then you end up in a [psycopg2.errors.ActiveSqlTransaction: CREATE DATABASE cannot run inside a transaction block](https://stackoverflow.com/questions/74899785/psycopg2-errors-activesqltransaction-create-database-cannot-run-inside-a-transa) error. You can try it out yourself ... this was the only way I was able to make it work

Reply:
Then I think it should at elast be run in a try-finally block